plugins/fileservices/fileservices.cpp

```diff
@@ -3008,3 +3010,31 @@ FILESERVICES_API void FILESERVICES_CALL fsGetDropZones(ICodeContext *ctx, size32
     __lenResult = mb.length();
     __result = mb.detach();
 }
+
+
+
+FILESERVICES_API int FILESERVICES_CALL fsGetExpireDays(ICodeContext * ctx, const char *_lfn)
+{
+    StringBuffer lfn;
+    constructLogicalName(ctx, _lfn, lfn);
+    Linked<IUserDescriptor> udesc = ctx->queryUserDescriptor();
+    Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(lfn.str(),udesc, false,false, false, nullptr, defaultPrivilegedUser);
+    int expireDays = -1;
+    if (df)
+        df->getExpire(expireDays);
+    else
+        throw MakeStringException(0, "GetExpireDays: Could not find logical file %s", lfn.str());
+    return expireDays;
+}
+
+FILESERVICES_API void FILESERVICES_CALL fsSetExpireDays(ICodeContext * ctx, const char *_lfn, int expireDays)
+{
+    StringBuffer lfn;
+    constructLogicalName(ctx, _lfn, lfn);
+    Linked<IUserDescriptor> udesc = ctx->queryUserDescriptor();
+    Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(lfn.str(),udesc, false,false, false, nullptr, defaultPrivilegedUser);
+    if (df)
+        df->setExpire(expireDays);
+    else
+        throw MakeStringException(0, "GetExpireDays: Could not find logical file %s", lfn.str());
```

Comment:
trivial: MakeStringException should be deprecated really, makeStringExceptionV is preferable

Reply:
Changed